tests/tree_util_tests.py

```diff
@@ -124,6 +124,19 @@ def tree_unflatten(cls, meta, data):
     (FlatCache({"a": [1, 2]}),),
 )
 
+try:
+  from dataclasses import dataclass  # pylint: disable=g-import-not-at-top
+
+  @dataclass
+  class ADataClass:
+    foo: object
+    bar: object
+
+  TREES += (ADataClass(foo="hello", bar=3.5),),
+  TREES += (ADataClass(foo="hello", bar=(ADataClass(foo=3.5, bar="quux"))),),
+except ModuleNotFoundError:
+  pass
+
```

Comment:
Ths looks pretty good. I prefer to keep the `try` block as succinct as possible to prevent catching unintended errors; you could do that with an `else` block:
```python
try:
  from dataclasses import dataclass  # pylint: disable=g-import-not-at-top
except ModuleNotFoundError:
  pass
else:
  @dataclass
  class ADataClass:
    foo: object
    bar: object

  TREES += (ADataClass(foo="hello", bar=3.5),),
  TREES += (ADataClass(foo="hello", bar=(ADataClass(foo=3.5, bar="quux"))),),
```

Reply:
fixed, thanks